src/python/pants/option/global_options.py

```diff
@@ -414,6 +414,16 @@ def register_bootstrap_options(cls, register):
             help="Read additional specs from this file (e.g. target addresses or file names). "
             "Each spec should be one per line.",
         )
+
+        register(
+            "--query",
+            type=list,
+            default=[],
+            metavar="<query-expr>",
+            help="A list of query expressions which process the input target specs in a "
+            "pipeline in order.",
```

Comment:
This should probably include a link to a page (maybe a section of https://www.pantsbuild.org/docs/advanced-target-selection or https://www.pantsbuild.org/docs/project-introspection) on the docsite that explains how to use query.

Reply:
I'd recommend this go into Advanced Target Selection. Maybe have a tooltip on Project Introspection that cross-references the other page.

You'll want to use https://www.pantsbuild.org/v2.0/docs/advanced-target-selection for now because this feature isn't being added to Pants 1.30.